src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -208,6 +218,37 @@ public RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(List<String> ca
         }
     }
 
+    private RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(RoleAssignmentQuery roleAssignmentQuery) {
+        RoleAssignmentResponse roleAssignmentResponse = new RoleAssignmentResponse();
+        roleAssignmentResponse.setRoleAssignments(new ArrayList<>());
```

Comment:
Could we please construct the array list with an allocated size of totalRecords, once it's retrieved from initial response? This will help help better memory management and performance.

Reply:
Done